src/interpolation/details/ArborX_InterpDetailsCompactRadialBasisFunction.hpp

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+#define ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Interpolation
+{
+
+#define CRBF_DECL(NAME)                                                        \
+  namespace Details                                                            \
+  {                                                                            \
+  template <std::size_t>                                                       \
+  struct NAME;                                                                 \
+  }                                                                            \
+                                                                               \
+  namespace CRBF                                                               \
+  {                                                                            \
+  template <std::size_t I>                                                     \
+  static constexpr Details::NAME<I> NAME{};                                    \
+  }
```

Comment:
What is this used for?
```suggestion
```

Reply:
Same reason as the `PolynomialDegree`, to be used as a function argument and help with templates.